hq/test/logic/IamRemediationTest.scala

```diff
@@ -144,7 +145,25 @@ class IamRemediationTest extends FreeSpec with Matchers {
   }
 
   "lookupCredentialId" - {
-    "TODO" ignore {}
+    val nonMatchingAccessKey = CredentialMetadata("adam.fisher", "AKIAIOSFODNN1EXAMPLE", date.minusDays(1), CredentialActive)
+    val matchingAccessKey = CredentialMetadata("amina.adewusi", "AKIAIOSFODNN2EXAMPLE", date, CredentialActive)
+
+    "given a key creation date matches a date in the metadata, return the correct metadata" in {
+      val result = lookupCredentialId(date, List(matchingAccessKey, nonMatchingAccessKey))
+      result.value.username shouldEqual "amina.adewusi"
```

Comment:
```suggestion
      result.value.username shouldEqual matchingAccessKey.username
```
(minor, but slightly less fragile if for whatever reason we update the test data)

Reply:
Sounds good 👍🏽